OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java

```diff
@@ -2305,7 +2305,7 @@ public void run() {
       if (shouldLogUserPrivacyConsentErrorMessageForMethodName(OSTaskController.SET_SUBSCRIPTION))
          return;
 
-      getCurrentSubscriptionState(appContext).setUserSubscriptionSetting(disable);
+      getCurrentSubscriptionState(appContext).setPushDisabled(!disable);
```

Comment:
Remove `!` we shouldn't be flipping the logic here. Your changes in `OSSubscriptionState.java` already account for this.